pkgs/by-name/fi/firezone-server/package.nix

```diff
@@ -0,0 +1,90 @@
+{
+  lib,
+  nixosTests,
+  fetchFromGitHub,
+  beamPackages,
+  gitMinimal,
+  pnpm_9,
+  nodejs,
+  tailwindcss_3,
+  esbuild,
+
+  mixReleaseName ? "domain", # "domain" "web" or "api"
+}:
+
+beamPackages.mixRelease rec {
+  pname = "firezone-server-${mixReleaseName}";
+  version = "0-unstable-2025-03-15";
+
+  src = "${
+    fetchFromGitHub {
+      owner = "firezone";
+      repo = "firezone";
+      rev = "09fb5f927410503b0d6e7fc6cf6a2ba06cb5a281";
+      hash = "sha256-1CZBFhOwX0DfXykPQ9tzn4tHg2tSnByXEPtlZleHK5k=";
+
+      # This is necessary to allow sending mails via SMTP, as the default
+      # SMTP adapter is current broken: https://github.com/swoosh/swoosh/issues/785
+      postFetch = ''
+        ${lib.getExe gitMinimal} -C $out apply ${./0000-add-mua.patch}
+      '';
```

Comment:
Does this need to be done in the fetcher? e.g. could it be a regular patch instead?

Reply:
Unfortunately yes, because this needs to be applied before the dependencies are fetched by `fetchMixDeps`